### ISSUE TITLE
Buttons do not have  an accessible name

### ISSUE THREAD

**Issue by tassoman** [vuejs/vitepress#2096]
### Describe the bug

I've ran _Crome's Lighthouse_ accessibility audit against the page, and it's failing, for the element marked as: `button.VPSwitch.VPSwitchAppearance` the item was recently reviewed for a11y and code (alpha58) compiles it inside a label.

Lighthouse tests expects the tag button has an [explicit](https://dequeuniversity.com/rules/axe/4.4/button-name) `aria-label` attribute. I can't say if it's false positive but affects the report.

### Reproduction

1. Setup a basic site
2. Open Chrome browser (v.100.x)
3. Open the basic website and browser's console
4. Inside Tab Lighthouse (9.4.0), check Accessibility and Desktop mode radio button 
5. Run Generate Report
6. See: warning Buttons do not have  an accessible name marked for button.VPSwitch.VPSwitchAppearance

### Expected behavior

Any warning expected for a11y tests.

### System Info

```shell
System:
    OS: Linux 5.10 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
    Memory: 11.43 GB / 12.31 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.15.1/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v16.15.1/bin/npm
  npmPackages:
    vitepress: ^1.0.0-alpha.58 => 1.0.0-alpha.58
```

### Additional context

related to #2080

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
<img width="320" alt="image" src="https://github.com/vuejs/vitepress/assets/40380293/6dbcef41-0e14-4b6c-99b5-f6a377c7d053">
